### PR TITLE
MODLOGIN-77: Use EventsLogAPI

### PR DESCRIPTION
**Pull request by opochepam**
Log `event type`, `tenant id`, `user id`, `ip`, `browser information` and `request timestamp`
while following events:

- **PASSWORD_RESET**
- **PASSWORD_CREATE**
- **PASSWORD_CHANGE**
- **PASSWORD_SUCCESSFUL_LOGIN_ATTEMPT**
- **FAILED_LOGIN_ATTEMPT**
- **USER_BLOCK**

Jira link: [MODLOGIN-77](https://issues.folio.org/browse/MODLOGIN-77)

### CODE REVIEW

File: ramls/logEvent.json [hunk 1]
```diff
@@ -8,14 +8,16 @@
...
+        "PASSWORD_RESET",
+        "PASSWORD_CREATE",
+        "PASSWORD_CHANGE",
+        "SUCCESSFUL_LOGIN_ATTEMPT",
+        "FAILED_LOGIN_ATTEMPT",
+        "USER_BLOCK"
+      ],
+      "description": "Event type"
     },
     "tenant": {
```
Comment (dmytrotur) [inline ramls/logEvent.json]: 'tenant' property is probably unnecessary is this schema as storage is tenant-specific.

File: src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java [hunk 2]
```diff
@@ -259,21 +267,34 @@ private void findUserCredentialById(String tenant, String token, String okapiUrl
...
-          saveUserCredential(pgClient, beginTx, asyncHandler, actionId, userCredential);
+          saveUserCredential(requestHeaders, beginTx, asyncHandler, actionId, userCredential);
         } else {
           Credential userCredential = createCredential(newPassword, credentialOpt.get());
           updateCredAndCredHistory(beginTx, userCredential, tenant, token, okapiUrl)
-            .setHandler(v -> deletePasswordActionById(pgClient, beginTx, asyncHandler, actionId, false));
+            .setHandler(v -> {
+              deletePasswordActionById(pgClient, beginTx, asyncHandler, actionId, false);
+
+              LogEvent logEvent = new LogEvent();
```
Comment (kostyantyn-kh) [inline src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java]: Probably it's better to create separate method for log action

File: src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java [hunk 3]
```diff
@@ -283,6 +304,15 @@ private void saveUserCredential(PostgresClient pgClient, AsyncResult<SQLConnecti
               asyncHandler.handle(Future.failedFuture(rollbackTx.cause())));
           return;
         }
+        LogEvent logEvent = new LogEvent();
+        logEvent.setEventType(LogEvent.EventType.PASSWORD_CREATE);
```
Comment (kostyantyn-kh) [inline src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java]: same here, separate method

File: src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java [hunk 4]
```diff
@@ -389,6 +419,15 @@ public PasswordStorageService updateCredential(JsonObject credJson, Map<String,
             if (v.failed()) {
               asyncResultHandler.handle(Future.failedFuture(v.cause()));
             } else {
+              LogEvent logEvent = new LogEvent();
+              logEvent.setEventType(LogEvent.EventType.PASSWORD_CHANGE);
```
Comment (kostyantyn-kh) [inline src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java]: and here

**Pull request merged**